system-parachains/asset-hubs/asset-hub-polkadot/tests/tests.rs

```diff
@@ -954,3 +954,291 @@ fn governance_authorize_upgrade_works() {
+			assert_relative_eq!(to_treasury as f64, (49_320 * UNITS) as f64, max_relative = 0.001);
+			// Total per day is ~328,797 DOT
+			assert_relative_eq!(
+				(to_stakers as f64 + to_treasury as f64),
+				(328_797 * UNITS) as f64,
+				max_relative = 0.001
+			);
+		});
+	}
+
+	#[test]
+	fn staking_inflation_correct_longer_era() {
+		ExtBuilder::<Runtime>::default()
+		.build()
+		.execute_with(|| {
+			// Twice the era duration means twice the emission:
+			let (to_stakers, to_treasury) = <staking::EraPayout as EraPayout<Balance>>::era_payout(
+				123, // ignored
+				456, // ignored
+				2 * MILLISECONDS_PER_DAY,
+			);
+
+			assert_relative_eq!(
+				to_stakers as f64,
+				(279_477 * UNITS) as f64 * 2.0,
+				max_relative = 0.001
+			);
+			assert_relative_eq!(
+				to_treasury as f64,
+				(49_320 * UNITS) as f64 * 2.0,
+				max_relative = 0.001
+			);
+		});
+	}
+
+	#[test]
+	fn staking_inflation_correct_whole_year() {
+		ExtBuilder::<Runtime>::default()
+		.build()
+		.execute_with(|| {
+			let (to_stakers, to_treasury) = <staking::EraPayout as EraPayout<Balance>>::era_payout(
+				123,                                  // ignored
+				456,                                  // ignored
+				(36525 * MILLISECONDS_PER_DAY) / 100, // 1 year
+			);
+
+			// Our yearly emissions is about 120M DOT:
+			let yearly_emission = 120_093_259 * UNITS;
+			assert_relative_eq!(
+				to_stakers as f64 + to_treasury as f64,
+				yearly_emission as f64,
+				max_relative = 0.001
+			);
+
+			assert_relative_eq!(to_stakers as f64, yearly_emission as f64 * 0.85, max_relative = 0.001);
+			assert_relative_eq!(
+				to_treasury as f64,
+				yearly_emission as f64 * 0.15,
+				max_relative = 0.001
+			);
+		});
+	}
+
+	// 10 years into the future, our values do not overflow.
+	#[test]
+	fn staking_inflation_correct_not_overflow() {
+		ExtBuilder::<Runtime>::default()
+		.build()
+		.execute_with(|| {
+			let (to_stakers, to_treasury) = <staking::EraPayout as EraPayout<Balance>>::era_payout(
+				123,                                 // ignored
+				456,                                 // ignored
+				(36525 * MILLISECONDS_PER_DAY) / 10, // 10 years
+			);
+			let initial_ti: i128 = 15_011_657_390_566_252_333;
+			let projected_total_issuance = (to_stakers as i128 + to_treasury as i128) + initial_ti;
+
+			// In 2034, there will be about 2.7 billion DOT in existence.
+			assert_relative_eq!(
+				projected_total_issuance as f64,
+				(2_700_000_000 * UNITS) as f64,
+				max_relative = 0.001
+			);
+		});
+	}
+
+	// Sets the view of the relay chain block number.
+	fn set_relay_number(n: RC_BlockNumber) {
+		ValidationData::<Runtime>::set(
+			Some( PersistedValidationData {
+				parent_head: vec![].into(),
+				relay_parent_number: n,
+				max_pov_size: Default::default(),
+				relay_parent_storage_root: Default::default(),
+			})
+		);
+	}
+
+	const TWO_YEAR_STEP_RATE: f64 = 0.2628;
```

Comment:
Maybe we can express this as fraction with `Perbill::from_rational(2628, 1000)`. Floats will otherwise have small errors already when defining the constants.